File: chatapp.client/src/app/models/message-view-model.ts
import { MessageMediaViewModel } from "./message-media-view-model";

export class MessageViewModel {
    id: number = 0;
    senderId: number = 0;
    recipientId: number = 0;
    content: string = '';
    media: MessageMediaViewModel | null = null;
    type: string = '';
    hasMedia: boolean = false;
    isSeen: boolean = false;
    parentMessageId: number = 0;
    createdAt: string = '';
    modifiedAt: string = '';
}

File: ChatApp.Server/Services/ViewModels/Messages/MessageViewModel.cs
﻿using ChatApp.Server.Services.ViewModels.Media;

namespace ChatApp.Server.Services.ViewModels.Messages
{
    public class MessageViewModel
    {
        public int Id { get; set; }

        public int SenderId { get; set; }

        public int RecipientId { get; set; }

        public string Content { get; set; } = "";

        public bool HasMedia { get; set; }

        public MediaViewModel? Media { get; set; }

        public bool IsSeen { get; set; }

        public int? ParentMessage { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime ModifiedAt { get; set; }
    }
}